web/src/lib/components/shared-components/change-location.svelte

```diff
@@ -66,24 +66,35 @@
     return `${name}${admin1Name ? ', ' + admin1Name : ''}${admin2Name ? ', ' + admin2Name : ''}`;
   };
 
-  const handleSearchPlaces = async () => {
-    if (searchWord === '' || isSearching) {
+  const handleSearchPlaces = () => {
+    if (searchWord === '') {
       return;
     }
 
-    // TODO: refactor setTimeout/clearTimeout logic - there are no less than 12 places that duplicate this code
-    isSearching = true;
-    const timeout = setTimeout(() => (showSpinner = true), timeBeforeShowLoadingSpinner);
-    try {
-      places = await searchPlaces({ name: searchWord });
-    } catch (error) {
-      places = [];
-      handleError(error, "Can't search places");
-    } finally {
-      clearTimeout(timeout);
-      isSearching = false;
-      showSpinner = false;
+    if (latestSearchTimeout) {
+      clearTimeout(latestSearchTimeout);
     }
+    showSpinner = true;
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    const searchTimeout = window.setTimeout(async () => {
+      try {
+        const searchResult = await searchPlaces({ name: searchWord });
+
+        // skip result when a newer search is happening
+        if (latestSearchTimeout === searchTimeout) {
+          places = searchResult;
+          showSpinner = false;
+        }
+      } catch (error) {
+        // skip error when a newer search is happening
+        if (latestSearchTimeout === searchTimeout) {
+          places = [];
+          handleError(error, "Can't search places");
+          showSpinner = false;
+        }
+      }
+    }, timeDebounceOnSearch);
```

Comment:
you should just ignore errors with eslint-disable. you probably need to write the code differently. maybe something like:
```suggestion
    const searchTimeout = window.setTimeout(() => {
      searchPlaces({ name: searchWord }).then(
        (searchResult) => {
          // skip result when a newer search is happening
          if (latestSearchTimeout === searchTimeout) {
            places = searchResult;
            showSpinner = false;
          }
        },
        (error) => {
          // skip error when a newer search is happening
          if (latestSearchTimeout === searchTimeout) {
            places = [];
            handleError(error, "Can't search places");
            showSpinner = false;
          }
        },
      );
    }, timeDebounceOnSearch);
```

Reply:
Other parts of code use await/async style so I just followed. I modified since you prefer this way.

Reply:
There's generally no problem with async/await. The issue here was that `window.setTimeout` expects you to pass it a synchronous function and you were passing it an async function